tox.ini

```diff
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{37,38,39,linting}
+    py{38,39,310,311,312,linting},mypy
```

Comment:
Note that this `mypy` environment is not running in CI.

I suggest actually that we run this in CI by configuring it in `.pre-commit-config.yaml` instead. Here's how pytest does it:

https://github.com/pytest-dev/pytest/blob/8fb7e8b31efaa55e760c142e26eb82b42081ca28/.pre-commit-config.yaml#L60-L72

Reply:
Good catch @nicoddemus